hedera-node/src/main/java/com/hedera/services/store/contracts/HederaWorldState.java

```diff
@@ -85,93 +81,44 @@ public HederaWorldState(
-				// selfdestruct adjusts balances to zero
-				account.setBalance(BigInteger.ZERO);
-				validateTrue(!repositoryRoot.isDeleted(address.toArray()), FAIL_INVALID);
-				account.setDeleted(true);
-			}
-		});
+	public List<ContractID> persist() {
+		repositoryRoot.flush();
 
-		final var createdContracts = new ArrayList<ContractID>();
+		final var copy = new LinkedList<>(privisionalContractCreations);
+		privisionalContractCreations.clear();
+
+		return copy;
+	}
 
+	@Override
+	public void customizeSponsoredAccounts() {
 		// copy over sponsor account info for CREATE operations
 		sponsorMap.forEach((contract, sponsorAddress) -> {
-			if (!provisionalAccountUpdates.containsKey(contract)) return;
-
+			AccountID newlyCreated = accountParsedFromSolidityAddress(contract.toArray());
 			AccountID sponsorAccount = accountParsedFromSolidityAddress(sponsorAddress.toArrayUnsafe());
-
-			WorldStateAccount provisionalAccount = provisionalAccountUpdates.get(contract);
-			if (ledger.exists(sponsorAccount)) {
-				MerkleAccount sponsorMerkleAccount = ledger.get(sponsorAccount);
-				provisionalAccount.setProxyAccount(sponsorMerkleAccount.getProxy());
-				provisionalAccount.setAutoRenew(sponsorMerkleAccount.getAutoRenewSecs());
-				provisionalAccount.setExpiry(sponsorMerkleAccount.getExpiry());
-			} else if (provisionalAccountUpdates.containsKey(sponsorAddress)) {
-				WorldStateAccount sponsorProvisionalAccount = provisionalAccountUpdates.get(sponsorAddress);
-				provisionalAccount.setProxyAccount(sponsorProvisionalAccount.getProxyAccount());
-				provisionalAccount.setAutoRenew(sponsorProvisionalAccount.getAutoRenew());
-				provisionalAccount.setExpiry(sponsorProvisionalAccount.getExpiry());
-			}
-		});
-
-		provisionalAccountUpdates.forEach((address, worldStateAccount) -> {
-			final var accountId = accountParsedFromSolidityAddress(address.toArray());
-			AccountState updatedAccount;
-
-			if (ledger.exists(accountId)) {
-				updatedAccount = repositoryRoot.getAccount(address.toArray());
-			} else {
-				updatedAccount = repositoryRoot.createAccount(address.toArray());
-				updatedAccount.setSmartContract(true);
-				updatedAccount.setExpirationTime(worldStateAccount.getExpiry());
-				updatedAccount.setAutoRenewPeriod(worldStateAccount.getAutoRenew());
-				updatedAccount.setProxyAccountShard(worldStateAccount.getProxyAccount().shard());
-				updatedAccount.setProxyAccountRealm(worldStateAccount.getProxyAccount().realm());
-				updatedAccount.setProxyAccountNum(worldStateAccount.getProxyAccount().num());
-				if (provisionalAccountDeletes.contains(address)) {
-					updatedAccount.setDeleted(true);
-				}
-				createdContracts.add(EntityIdUtils.asContract(accountId));
-			}
-
-			updatedAccount.setBalance(worldStateAccount.getBalance().getAsBigInteger());
-			ledgerAccountsSource.put(address.toArray(), updatedAccount);
+			validateTrue(ledger.exists(newlyCreated), FAIL_INVALID);
+			validateTrue(ledger.exists(sponsorAccount), FAIL_INVALID);
```

Comment:
I think that we will throw `FAIL_INVALID` if the sponsor account was deleted as part of the TX (suicide) after he created the new contract. What if we do not check for `exists` (for the sponsor account)? Are there going to be issues with that approach?

Reply:
Discussed it offline. This comment is not valid